src/app/view-config-utils.js

```diff
@@ -201,10 +203,21 @@ export function checkTypes(config) {
  * @param {object} config The view config prop.
  */
 export function initialize(config) {
-  if (config.initStrategy === 'auto') {
-    return initializeAuto(config);
+  let newConfig = cloneDeep(config);
+  if (newConfig.initStrategy === 'auto') {
+    newConfig = initializeAuto(config);
   }
-  return config;
+  const { layout } = newConfig;
+  layout.forEach((view, i) => {
+    // Assign uids for views where they are not present.
+    if (!view.uid) {
+      layout[i].uid = uuidv4();
+    }
+  });
+  return {
+    ...newConfig,
+    layout,
+  };
```

Comment:
why not make this part of `auto`?

Reply:
I think the views need `uid` values regardless of the initialization strategy, since in this PR I switched to using them in the removeComponent function.